### ISSUE TITLE
Improving typing for getPrototypeListOf

### ISSUE THREAD

**Issue by AbdealiLoKo** [fasttime/Polytype#17]
I was using `super.class(BaseClass)` in may cases and it handles typing correctly.
I see: `protected class<U extends T>(type: U): U;` in the `.d.ts` file

I was trying out `getPrototypeListOf` and realized that I was getting all the baseClass as `any`
And I saw: `export function getPrototypeListOf(o: any): any[];` in the `.d.ts` file
So, creating this issue to potentially improve the typing here to get more accurate types

PS: I have no idea if it is possible at all :) 

I did try this:
```ts
export class ChildClass extends classes(BaseClass1, BaseClass2) {
    myFunc() {
        (getPrototypeListOf(ChildClass) as (BaseClass1 | BaseClass2)[]).forEach(baseClass => {
            baseClass.myFunc()
        })
    }
}
```
NOTE: This will assume  `baseClass` has the type `(typeof BaseClass1 | typeof BaseClass2)`
So, if all my base classes implement a particular function - then this work out quite well

Doing: `getPrototypeListOf(ChildClass) as [BaseClass1, BaseClass2]` would also work and be more accurate

**Comment by fasttime** [comment]
Yes this would be helpful in many cases, but I would not want to change the type definition of `getPrototypeListOf` for this.
The current typing is intended to keep `getPrototypeListOf` consistent with `Object.getPrototypeOf`, which returns `any`.
Crucially, `Object.getPrototypeOf` and its counterpart `Object.setPrototypeOf` are inherently not type-safe, because they are intended to work with objects whose type changes dynamically or is not known at compile time (like setting the prototype or getting the prototype to access its properties).
But there could certainly be separate helper types and functions to handle what is not trivial using the core functionality.

In the case you show, since the list of base classes is known in advance, one could even rewrite the code without `getPrototypeListOf` like this:

```js
const baseClasses = [BaseClass1, BaseClass2];
export class ChildClass extends classes(...baseClasses) {
    myFunc() {
        baseClasses.forEach(baseClass => {
            baseClass.prototype.myFunc();
        })
    }
}
```

(Note that what is named `baseClass` in your for-each is actually a prototype)
